#### Q1: What is the difference between a t-test and a z-test? Provide an example scenario where you would use each type of test

Ans:

T test and Z test both is used in hypothesis testing for making conclusion.

T test used in smaller sample size and z test used in large sample size.

When in problem statment , we are not get population standerd deviation then we use T test.

when you know population standerd deviation and sample size is more than or above or equal to 30 then use z test.

when you know population standerd deviation but sample size is less than 30 then use T test.

when you unknow population standerd deviation and sample size is less than 30 then use T test.

z test = (sample_mean - population_mean) / (population std deviation)

T test = (sample_mean - population_mean) / (sample standerd deviation / np.sqrt(sample_size)

#### Q2: Differentiate between one-tailed and two-tailed tests.

Ans:

One-tailed Tests

A one-tailed test may be either left-tailed or right-tailed.

A left-tailed test is used when the alternative hypothesis states that the true value of 

the parameter specified in the null hypothesis is less than the null hypothesis claims.

A right-tailed test is used when the alternative hypothesis states that the true value of

the parameter specified in the null hypothesis is greater than the null hypothesis claims.


Two-tailed Tests

The main difference between one-tailed and two-tailed tests is that one-tailed tests will 

only have one critical region whereas two-tailed tests will have two critical regions. If we require a 100(1−α)

% confidence interval we have to make some adjustments when using a two-tailed test.

The confidence interval must remain a constant size, so if we are performing a two-tailed test,

as there are twice as many critical regions then these critical regions must be half the size. 

This means that when we read the tables, when performing a two-tailed test, we need to consider α2

rather than α.

#### Q3: Explain the concept of Type 1 and Type 2 errors in hypothesis testing. Provide an example scenario for each type of error.

Ans:

Hypothesis testing is evaluate the evidances for to reject or accept null hypothesis.

Its type of infrential statistics. It come on conclusion for making decision.

when you perform hypothesis testing , 

reality : null hypothesis is True ( accept ) and null hypothesis is False ( reject )
    
Decision : null hypothesis is True ( accept ) and null hypothesis is False( accept )
    
    
Type 1 Error :
    
    we reject null hypothesis when in reality null hypothesis is True.
    
    that is False positive condition.
    
    eg :you have data of iris setosa and iris virginica.
        
        when your actual value is setosa but it is predicted as virginica.
        
        
        
Type 2 Error :
    
    we accept null hypothesis when in reality it is False.
    
    that is False Negative condition.
    
    eg : your actual value is virginica but it is predicted as setosa.

#### Q4: Explain Bayes's theorem with an example

Ans:

Bayes statistics approach of data analysis and estimation parameter based on bayes theorem.

p(B/A) = p(B) * p(A/B) / p(A)


P(B/A)  = Likelihood probability : probability of B . A is happened.

p(A/B) = posterior probability : probability of A . B is happened.

p(A) = priority probability

p(B) = Marginal probaility



Likelihood probability = Marginal probability * posterior probability / prior probability.

#### Q5: What is a confidence interval? How to calculate the confidence interval, explain with an example.

Ans:

confidence interval is the range of values to estimate unknown population parameter.

confidence interval is also called interval estimate.


confidence interval = (sample mean) +  z * (standerd deviation / np.sqrt(sample size))

eg :

A retail company wants to rate the quality of a product by using results from a customer satisfaction survey. 

In the survey, the company asks respondents to rate the quality on a scale of one to five, 

with one being the poorest quality and five being the highest quality.

The sample size is 25, the sample mean is 4.5 and the standard deviation is 2.5. 

The company calculates the confidence interval assuming a 97% confidence level.

In [29]:
sample_size = 25

sample_mean = 4.5

standerd_deviation = 2.5

z = 0.97

import numpy as np
import math

In [30]:
confidence_interval_lower_limit = (sample_mean) - z * (standerd_deviation / np.sqrt(sample_size))

In [31]:
confidence_interval_lower_limit

4.015

In [32]:
confidence_interval_upper_limit = (sample_mean) + z * (standerd_deviation / np.sqrt(sample_size))

In [33]:
confidence_interval_upper_limit

4.985

conclusion :

The confidence interval ranges between 4.985 and 4.015, which means the customers in the sample rate the product quality between four and five. The company's confidence level of 97% represents the likelihood that these results are the same within the entire population of customers.

#### Q7. Calculate the 95% confidence interval for a sample of data with a mean of 50 and a standard deviation of 5. Interpret the results.

Ans:

In [34]:
z = 0.95

sample_mean = 50

standerd_deviation = 5

sample_size = 25

In [35]:
confidence_interval_lower_limit = (sample_mean) - z * (standerd_deviation/ np.sqrt(sample_size))

In [36]:
confidence_interval_lower_limit

49.05

In [37]:
confidence_interval_upper_limit = (sample_mean) + z * (standerd_deviation / np.sqrt(sample_size))

In [38]:
confidence_interval_upper_limit

50.95

confidence interval range between 49.05 to 50.95

#### Q8. What is the margin of error in a confidence interval? How does sample size affect the margin of error? Provide an example of a scenario where a larger sample size would result in a smaller margin of error.

Ans:

The margin of error is half the confidence interval. The larger the sample, the smaller the margin of error.

As sample size increases, the margin of error decreases. 

As the variability in the population increases, the margin of error increases.

As the confidence level increases, the margin of error increases.

#### Q9. Calculate the z-score for a data point with a value of 75, a population mean of 70, and a population standard deviation of 5. Interpret the results

Ans:

In [41]:
x = 75

pop_mean = 70

standerd_deviation = 5

In [42]:
z_score = (x - pop_mean) / standerd_deviation

In [43]:
z_score

1.0

#### Q10. In a study of the effectiveness of a new weight loss drug, a sample of 50 participants lost an average of 6 pounds with a standard deviation of 2.5 pounds. Conduct a hypothesis test to determine if the drug is significantly effective at a 95% confidence level using a t-test.

Ans:

In [54]:
n = 50

sample_mean = 6

standerd_deviation = 2.5

CI = 0.95

pop_mean = 0

significance_value = 1 - 0.95

In [55]:
significance_value

0.050000000000000044

Null hypothesis(H0):
    
    Assumption :  The weight loss drug is not significantly effective, and the average weight loss in the 
        
                  population is equal to or less than 0 pounds. 
            
            
                  The null hypothesis assumes that the population mean weight loss is equal to or less than 0 pounds.
                
                  In other words, the null hypothesis assumes that the weight loss drug has no effect on weight loss
    
Alternate Hypothesis(H1):
    
    Assumption : The weight loss drug is significantly effective, 
        
                 and the average weight loss in the population is greater than 0 pounds.
            
            
                The alternative hypothesis assumes that the population mean weight loss is greater than 0 pounds, 
                
                which would mean that the weight loss drug is effective in causing weight loss. 
                
                
                
This is a common assumption for null hypotheses in studies investigating the effectiveness of interventions or treatments, 

where the null hypothesis assumes no effect of the intervention or treatment.

In [57]:
t_test = (sample_mean - pop_mean ) / (standerd_deviation / np.sqrt(n))

In [58]:
t_test

16.970562748477143

In [60]:
df = n - 1

In [61]:
df

49

In [62]:
from scipy.stats import t

In [72]:
p_value = t.sf(abs(t_test) , df) 

In [73]:
p_value

1.8584420417635101e-22

In [76]:
if p_value < significance_value:
    print('reject the null hypothesis')
    
else:
    print('Fail to reject the null hypothesis')

reject the null hypothesis


conclusion :
    
    The population mean weight loss is greater than 0 pounds, 
                
    which would mean that the weight loss drug is effective in causing weight loss. 
                

#### Q11. In a survey of 500 people, 65% reported being satisfied with their current job. Calculate the 95% confidence interval for the true proportion of people who are satisfied with their job

In [109]:
n = 500

p = 0.65

z = 1.96 # ( z value for 95 % CI)

## p: proportion of people in the sample who are satisfied with their job (p-hat)

In [110]:
z

1.96

In [111]:
confidence_interval_higher = p + z *(np.sqrt (p *(1 - p) / n))

In [112]:
confidence_interval_higher

0.6918082288550951

In [113]:
confidence_interval_lower = p - z *(np.sqrt (p *(1 - p) / n))

In [114]:
confidence_interval_lower

0.608191771144905

conclusion : 
    
    Therefore, the 95% confidence interval for the true proportion of people who are satisfied with their job is 
    
    (0.608, 0.692) or (60.8%, 69.2%). This means that we can be 95% confident that the true proportion of people who
    
    are satisfied with their job lies between 60.8% and 69.2%.

#### Q13. A population has a mean of 60 and a standard deviation of 8. A sample of 50 observations has a mean of 65. Calculate the 90% confidence interval for the true population mean.

Ans:

In [149]:
sample_mean = 65

standerd_deviation = 8

n = 50

z = 1.645

significance_value = 1 - 0.90

In [133]:
CI_lower_limit = (sample_mean) - z *(standerd_deviation / np.sqrt(n))

In [134]:
CI_lower_limit

63.13889495191701

In [135]:
CI_higher_limit = (sample_mean) + z *(standerd_deviation / np.sqrt(n))

In [136]:
CI_higher_limit 

66.86110504808299

conclusion :The 90% confidence interval for the true population mean is between 63.13889495191701 to 66.86110504808299

#### Q14. In a study of the effects of caffeine on reaction time, a sample of 30 participants had an average reaction time of 0.25 seconds with a standard deviation of 0.05 seconds. Conduct a hypothesis test to determine if the caffeine has a significant effect on reaction time at a 90% confidence level using a t-test.

Ans:

Null hypothesis:
    
    Assumption : the caffeine has a not significant effect on reaction time
        
        
Alternate hypothesis :
    
    Assumption : the caffeine has a significant effect on reaction time not equal to 0.25 seconds
    
    

In [137]:
n = 30

sample_mean = 0.25

standerd_deviation = 0.05

pop_mean = 0

In [138]:
t_stat = (sample_mean - pop_mean) / (standerd_deviation / np.sqrt(n))

In [139]:
t_stat

27.386127875258307

In [145]:
df = n - 1

In [147]:
p_value = t.sf(abs(t_stat) ,df)

In [148]:
p_value

1.4162622442556677e-22

In [150]:
if p_value < significance_value:
    
    print('reject the null hypothesis')
    
else:
    
    print('Failed to reject the null hypothesis')

reject the null hypothesis


conclusion:
    
    Based on results the caffeine has a significant effect on reaction time at 90 % confidence interval